kotlintest-assertions-arrow/src/main/kotlin/io/kotlintest/assertions/arrow/either/matchers.kt

```diff
@@ -60,10 +60,11 @@ inline fun <reified A> beLeftOfType() = object : Matcher<Either<Any?, Any?>> {
         Result(false, "Either should be Left<${A::class.qualifiedName}> but was Right(${value.b})", "")
       }
       is Either.Left -> {
-        if (value.a is A)
+        val valueA = value.a
+        if (valueA is A)
           Result(true, "Either should be Left<${A::class.qualifiedName}>", "Either should not be Left")
         else
-          Result(false, "Either should be Left<${A::class.qualifiedName}> but was Left<${value.a::class.qualifiedName}>", "Either should not be Left")
+          Result(false, "Either should be Left<${A::class.qualifiedName}> but was Left<${if (valueA == null) "Null" else valueA::class.qualifiedName}>", "Either should not be Left")
```

Comment:
Can we replace the output with the usage of `convertValueToString` in kotlintest-assertions as this handles outputting null, empty string in a nicer way.

Reply:
I have looked into this, but I cannot work out how this could be used here. The potential nullable is used to get it's qualified class name. If I call `convertValueToString(value.a)` there is no way to get the qualified class name as it would return a `String`.